recipes/libheif/all/conanfile.py

```diff
@@ -0,0 +1,89 @@
+import os
+from conans import ConanFile, tools, CMake
+
+
+class Libheif(ConanFile):
+    name = "libheif"
+    description = "libheif is an HEIF and AVIF file format decoder and encoder."
+    topics = ("conan", "libheif", "heif", "codec", "video")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/strukturag/libheif"
+    license = ("LGPL-3.0-only", "GPL-3.0-or-later", "MIT")
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False],
+               "with_x265": [True, False]}
+    default_options = {"shared": False, "fPIC": True,
+                       "with_x265": False}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = '{}-{}'.format(self.name, self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def requirements(self):
+        self.requires("libde265/1.0.8")
+        if self.options.with_x265:
+            self.requires("libx265/3.2.1")
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        for target in ["X265", "Libde265", "LibAOM"]:
+            os.unlink(os.path.join(self._source_subfolder, "cmake", "modules", "Find"+target+".cmake"))
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["WITH_EXAMPLES"] = False
+        self._cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
```

Comment:
Since this is removed in the patch it should be handled by conan
```suggestion
```

Reply:
@SpaceIm based on your [rule of thumb](https://github.com/conan-io/conan-center-index/pull/3385#discussion_r516538705), there is no need in this line, right?

Reply:
Looking quickly at upstream CMakeLists.txt, there is no OBJECT or STATIC targets linked to potentially SHARED target, so indeed, this line is not required.